packages/discord.js/src/structures/InteractionCallback.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const { DiscordSnowflake } = require('@sapphire/snowflake');
+
+/**
+ * Represents an interaction callback response from Discord
+ */
+class InteractionCallback {
+  constructor(client, data) {
+    /**
+     * The client that instantiated this.
+     * @name InteractionCallback#client
+     * @type {Client}
+     * @readonly
+     */
+    Object.defineProperty(this, 'client', { value: client });
+
+    /**
+     * The id of the original interaction response
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The type of the original interaction
+     * @type {InteractionType}
+     */
+    this.type = data.type;
+
+    /**
+     * The id of the channel the original interaction was sent in
+     * @type {?Snowflake}
+     */
+    this.channelId = data.channel_id ?? null;
+
+    /**
+     * The id of the guild the original interaction was sent in
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id ?? null;
+
+    /**
+     * The instance id of the Activity if one was launched or joined
+     * @type {string}
+     */
+    this.activityInstanceId = data.activity_instance_id ?? null;
+
+    /**
+     * The id of the message that was created by the interaction
+     * @type {Snowflake}
+     */
+    this.responseMessageId = data.response_message_id ?? null;
+
+    /**
+     * Whether or not the message is in a loading state
+     * @type {boolean}
+     */
+    this.responseMessageLoading = data.response_message_loading ?? null;
+
+    /**
+     * Whether or not the response message was ephemeral
+     * @type {boolean}
+     */
+    this.responseMessageEphemeral = data.response_message_ephemeral ?? null;
```

Comment:
Am I right in thinking these effectively make `interaction.ephemeral` and `interaction.deferred` redundant?

Reply:
Sort of, though don't we use those internally for other purposes?

Reply:
Looks like we use them internally for checking if they are already in such state before responding with such a state. Example:

https://github.com/discordjs/discord.js/blob/3c74aa204909323ff6d05991438bee2c583e838b/packages/discord.js/src/structures/interfaces/InteractionResponses.js#L182-L190

Never mind then. Wishful thinking, perhaps...